src/Features/Core/Portable/ExternalAccess/VSTypeScript/Api/VSTypeScriptDiagnosticData.cs

```diff
@@ -30,6 +31,10 @@ public ImmutableArray<string> CustomTags
             => _data.CustomTags;
 
         public LinePositionSpan GetLinePositionSpan(SourceText sourceText, bool useMapped)
-            => DiagnosticData.GetLinePositionSpan(_data.DataLocation, sourceText, useMapped);
+        {
+            // TypeScript has no concept of mapped spans, so this should never be passed 'true'.
+            Contract.ThrowIfTrue(useMapped);
```

Comment:
I have checked all of TS's callsites of this helper, they always pass 'false' for 'useMapped'.

Reply:
Does webforms not go through this at all?  IIRC webforms supports JS script blocks

Reply:
Can we add [Obsolete] instead? Or is this easier for insertions?

Reply:
I don't see any use of this except in TS where they always pass 'false'. 

> Can we add [Obsolete] instead? Or is this easier for insertions?

yes.  let me do that.